src/nupic/algorithms/TemporalMemory.cpp

```diff
@@ -591,37 +425,58 @@ void TemporalMemory::compute(
                              activeColumnsUnsorted + activeColumnsSize);
   std::sort(activeColumns.begin(), activeColumns.end());
 
-  for (const ExcitedColumnData& excitedColumn : ExcitedColumns(activeColumns,
-                                                               activeSegments_,
-                                                               matchingSegments_,
-                                                               cellsPerColumn_))
+  const auto identity =
+    [](UInt c) { return c;};
```

Comment:
Explain what you mean? It's pretty typical to pass in an "identity" function into higher-order functions. I suppose I could stop using a lambda and instead use templates. i.e. `identity<UInt>` would be a function.